server/api/recipes.js

```diff
@@ -28,60 +28,148 @@ router.get('/:id', (req, res, next) => {
-  .then((data) => {
-    const title = data.properties.name[0];
-    const ingredientArr = data.properties.ingredient;
-    const imageUrl = data.properties.photo[0];
-    const recipePromise = Recipe.findOrCreate({
+  // console.log('req.body: ', req.body);
+  let recipePromise;
+  let groceryListPromise;
+  let ingredientArr;
+  let usr;
+
+  if (req.body.isFromChromeExt) {
+    const { name, email } = req.body.user;
+    // get user first
+    User.findOne({
       where: {
-        title,
-        // author: req.body.author,
-      },
-      defaults: {
-        recipeUrl: url,
-        imageUrl,
-        // siteName: req.body.siteName,
-        // tags: req.body.tags,
-        // isFavorite: req.body.isFavorite,
-        // numServings: req.body.numServings,
+        name: name.trim(),
+        email: email.trim(),
       },
-    });
-    // get grocery list
-    const groceryListPromise = req.user.getGrocerylist();
-    return Promise.all([recipePromise, groceryListPromise, ...ingredientArr])
-  })
-  .then(([recipeArr, grocerylist, ...ingredientArr]) => {
+    })
+    .then((user) => {
+      usr = user;
+      const { title, recipeUrl, imageUrl, author, siteName, numServings } = req.body.recipe;
+      recipePromise = Recipe.findOrCreate({
+        where: {
+          title,
+        },
+        defaults: {
+          recipeUrl,
+          imageUrl,
+          author,
+          siteName,
+          numServings,
+        },
+      });
+      ingredientArr = req.body.ingredients;
+      return (Promise.all([recipePromise, ...ingredientArr]));
+    })
+    .then(([recipeArr, ...arrIngredients]) => {
+      const newRecipe = recipeArr[0];
+      const isCreated = recipeArr[1];
+
+      // TODO: modularize to set following two associations - pass user argument to handle both post branches
+      usr.addSavedRecipes([newRecipe]);
+
+      usr.addGroceryListRecipe([newRecipe]);
+
+      // TODO: consider modularizing with adding passed argument if from microformat branch
+      if (isCreated) {
+        const arrIngredientPromises = arrIngredients.map((ingredient) => {
+          return Ingredient.findOrCreate({
+            where: {
+              name: ingredient.name,
+            },
+            defaults: {
+              unitMeasure: ingredient.unit,
+            },
+          })
+          .then(([foundIngredient, ingIsCreated]) => {
+            // TODO: only line different from microformat branch
+            foundIngredient.quantity = ingredient.quantity;
+            return foundIngredient;
+          })
+          .catch(next);
+        });
+
+        // TODO: modularize - same as in microformat branch
+        return Promise.all([newRecipe, ...arrIngredientPromises])
+          .then(([recipe, ...ingredients]) => {
+            const ingArr = recipe.addIngredients(ingredients);
+            return Promise.all([newRecipe, ingArr]);
+          })
+          .then(([recipe, ingredientsArr]) => Recipe.findById(recipe.id))
```

Comment:
maybe just send a status since the chrome extension has no use for the recipe obj